### DISCUSSION TITLE
Why are file endings eg .bam not shown in tracklist ?

### DISCUSSION THREAD

**Discussion by colindaven** [GMOD/jbrowse-components#2367] [Ideas]
Hi,

when using jbrowse2 more and more, we find it a bit strange that the file endings are not shown in the tracklist on the right.

Previously, we used jbrowse1 and the filter to just view all bams or vcfs in a project, for example.

Also, it's good to discern between bam, cram and bigwig files (often similarly named) in projects with moderate-large numbers of tracks.

```
#eg, copied from trackList: (.bam excluded)
B_S13_R1.trm.ns.fix.s
B_S13_R1.trm.ns.fix.s.mm.abra.calmd

```

Thanks,
Colin

**Comment by cmdcolin** [comment]
Just curious, are you referring to tracks loaded via the CLI or via the in-app mechanism? I think both remove the extension by default. We could consider keeping it by default in both maybe though

**Reply by colindaven** [reply]
This is all referring to the command line additions, I haven't really tried the in-app function yet. Would be great to keep the endings in there.

**Comment by garrettjstevens** [comment]
By default, the JBrowse CLI tries to guess a sensible name for the track based on the file name, which ends up being the file name without the final file extension. You can provide a custom track name with the `--name` flag, though. For example, if you wanted the track name to be the full file name, you could do something like:

```bash
MY_FILE=path/to/my/file.cram
jbrowse add-track ${MY_FILE} --load copy --name $(basename ${MY_FILE})
```

**Reply by colindaven** [reply]
That's an easy workaround, thanks, but I can't see how it is beneficial for anyone to remove this information. I know you can click on this to check the track type, but I am already dealing with confused users who used the .bam, .bw extensions in jbrowse1 widely. I would suggest retaining the file ending should be default behaviour. In the meantime I'll adapt my insert scripts.
